### PR TITLE
Add a GitHub action to run the CI tests

### PR DESCRIPTION
**Pull request by eatyourgreens**
Travis isn't running the CI tests for #1886, so this adds a GitHub actions script which installs and builds the monorepo packages, runs the tests, generates a coverage report and then posts to Coveralls.

This should, at least, allow me to tell whether the problem in #1886 is with the code or with Travis.

# Review Checklist

## General

- [ ] Are the tests passing locally and on Travis?
- [ ] Is the documentation up to date?

## Components
- [ ] Has a storybook story been created or updated?
- [ ] Is the component accessible? 
  - [ ] Can it be used with a screen reader? [BBC guide to testing with VoiceOver](https://bbc.github.io/accessibility-news-and-you/accessibility-and-testing-with-voiceover-os.html)
  - [ ] Can it be used from the keyboard? [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - [ ] Is it passing accessibility checks in the storybook?

## Apps

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `yarn panic && yarn bootstrap` or `docker-compose up --build` and app works as expected?

## Publishing

- [ ] Is the changelog updated?
- [ ] Are the dependencies updated for apps and libraries that are using the newly published library?

## Post-merging

- [ ] Did the app deploy to https://frontend.preview.zooniverse.org/projects/:project-name/:owner or https://frontend.preview.zooniverse.org/about?
- [ ] Is the new feature working or bug now fixed?
  - [ ] Is there a Talk or blog post written to announce the new feature(s)?
- [ ] Is the design working across browsers (Firefox, Chrome, Edge, Safari) and mobile?
  - [ ] Is this approved by our designer?
- [ ] Is this ready for production deployment?

### CODE REVIEW

File: .github/workflows/ci-tests.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+name: CI Tests
+
+on: [push]
+
+jobs:
+  release:
+   runs-on: ubuntu-latest
+   env:
+     CI: true
```
Comment (eatyourgreens) [inline .github/workflows/ci-tests.yml:9]: This is a placeholder. I'm not sure if we need to specify an environment for the test build.

File: .github/workflows/ci-tests.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+name: CI Tests
+
+on: [push]
```
Comment (camallen) [inline .github/workflows/ci-tests.yml:3]: I thin we want to modify this to mirror what we have in travis, pushes to master only and PR additions, 
- https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions/migrating-from-travis-ci-to-github-actions#targeting-specific-branches

- https://github.community/t/trigger-workflow-for-a-pull-request-and-for-all-subsequent-commits-in-it/17996/2
```suggestion
on:
  pull_request:
  push:
    branches:    
      - master
```
current travis config https://github.com/zooniverse/front-end-monorepo/blob/3ac16a3d1f7132fc1af42e2f793e331a8cd2e106/.travis.yml#L6-L8

File: .github/workflows/ci-tests.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+jobs:
+  release:
```
Comment (camallen) [inline .github/workflows/ci-tests.yml:6]: should this be named `build`? are we releasing?

File: package.json [hunk 4]
```diff
@@ -20,6 +20,7 @@
     "bootstrap": "./bin/bootstrap.sh",
     "build": "lerna run --parallel build",
     "coverage": "nyc report --reporter=text-lcov | coveralls",
+    "coverage-lcov": "nyc report --reporter=lcov",
```
Comment (camallen) [inline package.json:23]: Can we add coveralls secrets to push the results to coveralls? Or later PR ? 

Also OOI why the different reporter?

**Review by camallen — APPROVED** [review]
LGTM - suggestion on running on PR events and pushes to main (replicate travis setup) and question on coveralls.

Nice to see this setup 👍 🎉

File: package.json (hunk 4 above)
Comment (eatyourgreens) [reply package.json:23]: The Coveralls action looks for an LCOV file, then posts it using the `GITHUB_TOKEN` secret. This action writes the file.

File: .github/workflows/ci-tests.yml (hunk 3 above)
Comment (eatyourgreens) [reply .github/workflows/ci-tests.yml]: Good catch. I copied this across from the Sentry releases.

File: package.json (hunk 4 above)
Comment (camallen) [reply package.json:23]: Ah - and then the `Coveralls` step will post that file..gotcha 👍

Comment (eatyourgreens) [reply package.json:23]: I can remove the coverage script from Travis too. Right now it's posting twice. https://coveralls.io/repos/250962/builds

Comment (eatyourgreens) [reply package.json:22]: At least, I think it's posting twice. I might need to revisit this if the coverage stops updating.

**Pull request merged**